packages/backend/src/tests/ledgerTransfer.ts

```diff
@@ -0,0 +1,45 @@
+import { v4 as uuid } from 'uuid'
+import { TransactionOrKnex } from 'objection'
+import {
+  LedgerTransfer,
+  LedgerTransferState,
+  LedgerTransferType
+} from '../accounting/psql/ledger-transfer/model'
+
+interface CreateLedgerTransferArgs {
+  transferRef?: string
+  amount?: bigint
+  ledger: number
+  creditAccountId: string
+  debitAccountId: string
+  state?: LedgerTransferState
+  type?: LedgerTransferType
+  expiresAt?: Date
+}
+
+export const createLedgerTransfer = async (
+  args: CreateLedgerTransferArgs,
+  knex: TransactionOrKnex
+): Promise<LedgerTransfer> => {
+  const {
+    amount,
+    transferRef,
+    ledger,
+    creditAccountId,
+    debitAccountId,
+    state,
+    type,
+    expiresAt
+  } = args
+
+  return LedgerTransfer.query(knex).insertAndFetch({
+    transferRef: transferRef ?? uuid(),
```

Comment:
```suggestion
    transferRef,
```
via the `$beforeInsert`